### PR TITLE
Add support for newobj on struct values.

### PR DESCRIPTION
**Pull request by Washi1337**

### CODE REVIEW

File: src/Platforms/Echo.Platforms.AsmResolver/Emulation/Dispatch/ObjectModel/NewObj.cs [hunk 1]
```diff
@@ -26,16 +26,23 @@ public override DispatchResult Execute(ExecutionContext context, CilInstruction
...
             // Pop arguments.
             int argumentCount = environment.Architecture.GetStackPopCount(instruction);
             var arguments = context.ProgramState.Stack
                 .Pop(argumentCount, true)
                 .Cast<ICliValue>()
                 .ToList();
-            arguments.Insert(0, cilValueObject);
+
+            arguments.Insert(0, type.IsValueType
+                ? new OValue(cilValueObject, true, environment.Is32Bit)
```
Comment (zsr2531) [inline src/Platforms/Echo.Platforms.AsmResolver/Emulation/Dispatch/ObjectModel/NewObj.cs:44]: Does this simulate boxing behaviour automatically? Or what is the reason for the value being wrapped in an `OValue`?

**Review by zsr2531 — CHANGES_REQUESTED** [review]

File: src/Platforms/Echo.Platforms.AsmResolver/Emulation/Dispatch/ObjectModel/NewObj.cs (hunk 1 above)
Comment (Washi1337) [reply src/Platforms/Echo.Platforms.AsmResolver/Emulation/Dispatch/ObjectModel/NewObj.cs:44]: Technically it is not an `O`, but an `&`. According to the ECMA-335, `newobj` can initialize both reference types as well as value types. It first allocates the object/struct somewhere, and then calls the constructor referenced in the operand, with argument 0 (= `this`) pointing to this allocated memory.

Comment (Washi1337) [reply src/Platforms/Echo.Platforms.AsmResolver/Emulation/Dispatch/ObjectModel/NewObj.cs:44]: We just don't have support for proper `&` pointers yet, and `O` is very similar to `&`.

**Review by zsr2531 — APPROVED** [review]
LGTM!

**Pull request merged**